platform/darwin/src/string_nsstring.mm

```diff
@@ -27,5 +27,24 @@
     return result;
 }
 
+std::string formatNumber(double number,
+                         const std::string& localeId,
+                         const std::string& currency,
+                         uint8_t minFractionDigits,
+                         uint8_t maxFractionDigits) {
+    NSNumberFormatter *_numberFormatter;
+    _numberFormatter = [[NSNumberFormatter alloc] init];
```

Comment:
Does this method always get called on the same thread? If so, we can initialize the object just once using `dispatch_once`, hold onto it statically, and modify the options each time the method is called. Otherwise, we’ll need to lazily create a different formatter for each combination of options.

Alternatively, we can drop down to CFNumberFormatter, which might not have the same overhead.

Reply:
Previously, I had suggest `_numberFormatter` as the variable name with the assumption that the variable would be global rather than local. The underscore is unnecessary for a local variable.

(Also, you can combine a local variable declaration and definition on the same line.)

Reply:
Thank you for the comments. I removed the underscore and combined the declaration and definition on the same line.

Regarding the thread question, I am not sure as I'm not yet very familiar with the code (and even less with iOS/macOS side of things). If you want me to make the other changes that you proposed I would appreciate some more guidance. Do you have any good example on how to use `dispatch_once`? Or is it better to use CFNumberFormatter?